pkg/clients/aws.go

```diff
@@ -690,18 +692,9 @@ func IsPolicyUpToDate(local, remote *string) bool {
 // CleanError Will remove the requestID from a awserr.Error and return a new awserr.
 // If not awserr it will return the original error
 func CleanError(err error) error {
-	if err == nil {
-		return err
-	}
-	// TODO cvodak revisit this
-	// var re *awshttp.ResponseError
-	// if errors.As(err, &re) {
-	//	log.Printf("requestID: %s, error: %v", re.ServiceRequestID(), re.Unwrap());
-	//  }
-	// if awsErr, ok := err.(awserr.Error); ok {
-	//	return awserr.New(awsErr.Code(), awsErr.Message(), nil)
-	//}
-	return err
+	var awsErr smithy.APIError
+	_ = errors.As(err, &awsErr)
+	return awsErr
```

Comment:
`errors.As` doesn't make any change if the `err` is `nil`, so that case should be safe. However, while I'm writing the tests per your suggestion, I noticed it omits any non-APIError because `awsErr` ends up being `nil` if `errors.As` returns false.

Reply:
I added a `if err == nil { return nil}`, too, because `errors.As` pays the cost of a few `reflect` calls before returning `nil` for that case.